src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -4667,6 +4667,38 @@ private void VisitInitializer(IOperation rewrittenTarget, ISymbolInitializerOper
             {
                 LeaveRegion();
             }
+	}
+
+        public override IOperation VisitEventAssignment(IEventAssignmentOperation operation, int? captureIdForResult)
```

Comment:
>VisitEventAssignment [](start = 35, length = 20)

Can we assert that these are statements, or can they occur in an expression context? If they can, consider adding a test for scenario like that. Like a couple of tests with and without a control flow in the instance.

Reply:
> Can we assert that these are statements, or can they occur in an expression context? 

I don't think we can assert these are statements. These are always parented by an `IExpressionStatementOperation` for non-error cases, but may be child of an `IInvalidOperation` for error cases, say for example:

```
using System;

class C
{
#pragma warning disable CS0067 // Event is unused.
    public event EventHandler MyEvent;

    void M(C c, EventHandler handler1)
    /*<bind>*/{
        (c.MyEvent += handler1) = 0;
    }/*</bind>*/
}
```

```
Block[B0] - Entry
    Statements (0)
    Next (Regular) Block[B1]
Block[B1] - Block
    Predecessors: [B0]
    Statements (1)
        IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null, IsInvalid) (Syntax: '(c.MyEvent  ... dler1) = 0;')
          Expression: 
            ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.Void, IsInvalid) (Syntax: '(c.MyEvent  ... ndler1) = 0')
              Left: 
                IInvalidOperation (OperationKind.Invalid, Type: System.Void, IsInvalid, IsImplicit) (Syntax: 'c.MyEvent += handler1')
                  Children(1):
                      IEventAssignmentOperation (EventAdd) (OperationKind.EventAssignment, Type: System.Void, IsInvalid) (Syntax: 'c.MyEvent += handler1')
                        Event Reference: 
                          IEventReferenceOperation: event System.EventHandler C.MyEvent (OperationKind.EventReference, Type: System.EventHandler, IsInvalid) (Syntax: 'c.MyEvent')
                            Instance Receiver: 
                              IParameterReferenceOperation: c (OperationKind.ParameterReference, Type: C, IsInvalid) (Syntax: 'c')
                        Handler: 
                          IParameterReferenceOperation: handler1 (OperationKind.ParameterReference, Type: System.EventHandler, IsInvalid) (Syntax: 'handler1')
              Right: 
                ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 0) (Syntax: '0')

    Next (Regular) Block[B2]
Block[B2] - Exit
    Predecessors: [B1]
    Statements (0)
```

>  If they can, consider adding a test for scenario like that. Like a couple of tests with and without a control flow in the instance.

I am not sure about the `instance` that you are referring to. Can you give an example unit test?

Reply:
> I am not sure about the instance that you are referring to. 

The receiver of the event access.

> Can you give an example unit test?

I don't have a concrete answer at the moment, try to see if we can get an IEventAssignmentOperation node as a child of some expression in some error situation. It is probably impossible, but I think worth exploring a little bit.  


---
In reply to: [188798815](https://github.com/dotnet/roslyn/pull/26847#discussion_r188798815) [](ancestors = 188798815)

Reply:
> The receiver of the event access.

Ah, I have tests for these cases: `EventAssignment_ControlFlowInEventReference`, `EventAssignment_ControlFlowInEventReference_StaticEvent`  and `EventAssignment_ControlFlowInEventReferenceAndHandler`

---
In reply to: [188802407](https://github.com/dotnet/roslyn/pull/26847#discussion_r188802407) [](ancestors = 188802407,188798815)

Reply:
>  if we can get an IEventAssignmentOperation node as a child of some expression in some error situation

Yes, it is possible for error cases. Let me add a unit test.

---
In reply to: [188802407](https://github.com/dotnet/roslyn/pull/26847#discussion_r188802407) [](ancestors = 188802407,188798815)

Reply:
Add couple of tests for C#. In VB, IEventAssignmentOperation is always a child of IExpressionStatement.

---
In reply to: [188803086](https://github.com/dotnet/roslyn/pull/26847#discussion_r188803086) [](ancestors = 188803086,188802407,188798815)